cns/service/main.go

```diff
@@ -1201,6 +1201,9 @@ func reconcileInitialCNSState(ctx context.Context, cli nodeNetworkConfigGetter,
 	}
 
 	logger.Printf("Retrieved NNC: %+v", nnc)
+	if !nnc.DeletionTimestamp.IsZero() {
```

Comment:
assuming there no weird kubernetes thing that sets a deletiontimestamp to `January 1, year 1, 00:00:00 UTC` just to make it non-nil to signal deletion?

Reply:
The timestamp is set on objects with finalizers to indicate that they are pending deletion to those things that own the finalizers. I don't think in practice that this is ever non-nil and zero